### PR TITLE
modules/SceNpTrophy: complete implement Get(Game/Trophy)Info

### PR DESCRIPTION
**Pull request by Zangetsu38**

### CODE REVIEW

File: vita3k/modules/SceNpTrophy/SceNpTrophy.cpp [hunk 1]
```diff
@@ -143,7 +148,7 @@ EXPORT(int, sceNpTrophyGetGameInfo, np::trophy::ContextHandle context_handle, Sc
             }
         }
 
-        // TODO: set title and description
+        details->size = sizeof(SceNpTrophyGameDetails);
```
Comment (scribam) [inline vita3k/modules/SceNpTrophy/SceNpTrophy.cpp:151]: This line should be removed as `details->size` is already set and checked.

File: vita3k/modules/SceNpTrophy/SceNpTrophy.cpp [hunk 2]
```diff
@@ -171,6 +176,7 @@ EXPORT(int, sceNpTrophyGetGameInfo, np::trophy::ContextHandle context_handle, Sc
         }
 
         data->progressPercentage = data->unlockedTrophies * 100 / context->trophy_count;
+        data->size = sizeof(SceNpTrophyGameData);
```
Comment (scribam) [inline vita3k/modules/SceNpTrophy/SceNpTrophy.cpp:179]: ditto

File: vita3k/modules/SceNpTrophy/SceNpTrophy.cpp [hunk 3]
```diff
@@ -221,13 +227,19 @@ EXPORT(int, sceNpTrophyGetTrophyInfo, np::trophy::ContextHandle context_handle,
         details->trophyId = trophy_id;
         details->trophyGrade = context->trophy_kinds[trophy_id];
         details->hidden = context->is_trophy_hidden(trophy_id);
-        // TODO: set groupId, name and description
+        std::string name, detail;
+        context->get_trophy_details(trophy_id, name, detail);
+        memcpy((char *)details->name, name.c_str(), name.size() + 1);
+        memcpy((char *)details->description, detail.c_str(), detail.size() + 1);
+        details->size = sizeof(SceNpTrophyDetails);
```
Comment (scribam) [inline vita3k/modules/SceNpTrophy/SceNpTrophy.cpp:234]: ditto

File: vita3k/modules/SceNpTrophy/SceNpTrophy.cpp [hunk 4]
```diff
@@ -221,13 +227,19 @@ EXPORT(int, sceNpTrophyGetTrophyInfo, np::trophy::ContextHandle context_handle,
...
+        // TODO: set groupId
     }
 
     if (data) {
         data->trophyId = trophy_id;
         data->unlocked = context->is_trophy_unlocked(trophy_id);
         data->timestamp.tick = context->unlock_timestamps[trophy_id];
+        data->size = sizeof(SceNpTrophyData);
```
Comment (scribam) [inline vita3k/modules/SceNpTrophy/SceNpTrophy.cpp:242]: ditto

**Review by scribam — CHANGES_REQUESTED** [review]

**Comment by Zangetsu38** [comment]
Done

File: vita3k/modules/SceNpTrophy/SceNpTrophy.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/modules/SceNpTrophy/SceNpTrophy.cpp]: i have done all change requuest

**Review by frangarcj — APPROVED** [review]

**Comment by Zangetsu38** [comment]
@scribam waiting your review :/

**Review by scribam — APPROVED** [review]
Skeleton Rider [PCSB00588] now displays the title and description of trophies in TTY 👍

**Pull request merged**